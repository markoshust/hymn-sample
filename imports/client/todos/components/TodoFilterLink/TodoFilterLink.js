import React, { PropTypes } from 'react';
import Link from 'client/core/components/Link';

const TodoFilterLink = ({
  children,
  isActive,
  onClick,
}) => (
  isActive ? (
    <span>{children}</span>
  ) : (
    <Link
      href="#"
      onClick={onClick}
    >
      {children}
    </Link>
  )
);

TodoFilterLink.propTypes = {
  children: PropTypes.node.isRequired,
  isActive: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};

export default TodoFilterLink;
