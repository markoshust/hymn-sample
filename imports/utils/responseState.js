const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
