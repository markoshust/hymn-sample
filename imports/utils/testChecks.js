import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
