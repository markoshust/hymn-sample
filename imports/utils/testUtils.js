import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
