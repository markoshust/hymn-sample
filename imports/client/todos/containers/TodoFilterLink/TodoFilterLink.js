import { connect } from 'react-redux';
import { todosSetVisibilityFilter } from '../../actions';
import TodoFilterLink from '../../components/TodoFilterLink';

const mapStateToProps = (state, props) => ({
  isActive: props.filter === state.visibilityFilter,
});

const mapDispatchToProps = (dispatch, props) => ({
  onClick: () => dispatch(todosSetVisibilityFilter(props.filter)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TodoFilterLink);
