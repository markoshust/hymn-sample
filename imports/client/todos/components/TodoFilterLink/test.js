import React from 'react';
import { renderCheck } from 'utils/testChecks';
import TodoFilterLink from './TodoFilterLink';

const onSubmit = jest.fn();

describe('<TodoFilterLink />', () => {
  it('renders correctly', () => {
    renderCheck(
      <TodoFilterLink onClick={onSubmit}>
        Test
      </TodoFilterLink>
    );
  });
});

describe('<TodoFilterLink />', () => {
  it('renders correctly when active', () => {
    renderCheck(
      <TodoFilterLink
        isActive
        onClick={onSubmit}
      >
        Test
      </TodoFilterLink>
    );
  });
});
