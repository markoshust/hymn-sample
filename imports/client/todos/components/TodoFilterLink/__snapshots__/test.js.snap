exports[`<TodoFilterLink /> renders correctly 1`] = `
<a
  href="#"
  onClick={[Function anonymous]}>
  Test
</a>
`;

exports[`<TodoFilterLink /> renders correctly when active 1`] = `
<span>
  Test
</span>
`;
